packages/palette/src/elements/SelectInput/SelectInput.tsx

```diff
@@ -0,0 +1,527 @@
+import composeRefs from "@seznam/compose-react-refs"
+import { themeGet } from "@styled-system/theme-get"
+import React, { createRef, useCallback, useMemo, useRef, useState } from "react"
+import styled, { css, ExecutionContext } from "styled-components"
+import { height as systemHeight } from "styled-system"
+import { useKeyboardListNavigation } from "use-keyboard-list-navigation"
+import { RequiredField } from "../../shared/RequiredField"
+import { useContainsFocus, usePosition, useWidthOf } from "../../utils"
+import { Box, BoxProps, splitBoxProps } from "../Box"
+import { Input, InputProps } from "../Input"
+import { Text } from "../Text"
+import { SELECT_INPUT_STATES } from "./tokens"
+
+/**
+ * The option structure for the list in the dropdown menu
+ *
+ * @interface SelectOption
+ * @property {string} `text` is the content that will be displayed as selected option
+ * @property {string} `name` is the content that will be displayed in the dropdown list
+ * @property {string} `value` is the value that will be passed to onSelect
+ */
+export interface SelectOption {
+  text: string
+  name: string
+  value: string
+}
+
+export interface SelectInputProps extends Omit<InputProps, "onSelect"> {
```

Comment:
I mostly copied the code from `PhoneInput` and added a few props for customization.